notebook/services/contents/filemanager.py

```diff
@@ -39,6 +39,29 @@
 
 _script_exporter = None
 
+# workaround nbformat bug:
+# nbformat save logic will mutate the notebook if it does not
+# have cells ids, so if we sign the notebook,
+# the signature of the saved notebook is changed.
+# we thus need to add cell ids to the notebook before signing.
+try:
+    from nbformat.validator import normalize
+except ImportError:
+    from nbformat.corpus.words import generate_corpus_id
+    from copy import deepcopy
+
+    def normalize(nbdict, version, version_minor):
+        nbdict = deepcopy(nbdict)
```

Comment:
Could we make the `deepcopy` conditional on version? Not sure how many pre-cell-id notebooks are out there but this would save some time/space for them.